### ISSUE TITLE
`spd-say` speaks but afterwards hangs indefinitely on Postmarket OS  (Alpine Linux)

### ISSUE THREAD

**Issue by C-Loftus** [brailcom/speechd#1022]
Thank you for your work on speechd, it is super great and I really appreciate it. I am tinkering with getting the Odilia screen reader to work on a mobile Linux OS (https://github.com/odilia-app/odilia/discussions/175) and was wondering if you had any insight on this.

### Steps to reproduce

1. Install Postmarket OS on a Pinephone (I imagine this is difficult to repro, sorry! )
    - For what it's worth, I have not installed or modified any other software; this is a totally stock install
    - PostmarketOS is just a derivative of Alpine Linux and is a fairly minimal install so may be missing some accessibility related depdencies
3. Install speech dispatcher from alpine repos `sudo apk add speech-dispatcher`
4. Run  `spd-say test` 
    - Worth noting that I did not edit any of the speechd configs, add any other modules, etc.
6. `spd-say` speaks `test` correctly with what I presume to be espeak-ng (?) but it hangs forever and never returns 

This issue persists after reboot

### Obtained behavior

`spd-say` speaks `test` correctly but it hangs forever and never return

### Expected behavior

I expected `spd-say` to exit after saying the message and not hang

### Behavior information

I have done all of the following below:

```
* Set `LogLevel` to 5 in your `/etc/speech-dispatcher/speechd.conf`  
* Set `Debug` to 1 in your `/etc/speech-dispatcher/yourspeechmodule.conf` (e.g. `espeak-ng.conf`)
* Restart your computer to make sure it gets taken into account
* Reproduce the issue
* Make a copy of the `speech-dispatcher.log` file, it is usually found in `/run/user/1000/speech-dispatcher/log`. Also make a copy of `yourspeechmodule.log` (e.g. `espeak-ng.log`).
* Set `LogLevel` back to 3 in your `/etc/speech-dispatcher/speechd.conf`
* Attach the `speech-dispatcher.log` and `yourspeechmodule.log` file to your bug report.
* Also attach your `speechd.conf` file and `yourspeechdmodule.conf` file.
```

[speech-dispatcher.log](https://github.com/user-attachments/files/19527935/speech-dispatcher.log)

[espeak-ng.log](https://github.com/user-attachments/files/19527937/espeak-ng.log)

[speechd.conf](https://github.com/user-attachments/files/19527940/speechd.conf.txt) _(had to give this the file 
the name speechd.conf.txt since github didn't allow an upload for just .conf for some reason)_

From what I can see in the logs, it seems like maybe there is a deadlock at the end? But just a guess

```
[Sun Mar 30 23:29:21 2025 : 997651] speechd:     wait_for_poll: activity in Speech Dispatcher
[Sun Mar 30 23:29:21 2025 : 997683] speechd:     Locking element_free_mutex in speak()
[Sun Mar 30 23:29:21 2025 : 997710] speechd:     No message in the queue
```

### Distribution

```
uname -a


Linux pine64-pinephone 6.12.1 #2-postmarketos-allwinner SMP PREEMPT Sat Dec 21 00:33:52 UTC 20 aarch64 Linux
```

### Version of Speech-dispatcher

```
speech-dispatcher --version
speech-dispatcher 0.11.5
```

**Comment by sthibaul** [comment]
Would you be able to test 0.12 ? The output processing has been revamped there.

**Comment by C-Loftus** [comment]
Yes I can take a look. I presume I would need to compile from source so this may take me a bit but I can look into it.

If you have any other thoughts or guesses what might be going wrong as I look into this, feel free to let me know here. No worries if not. Thank you again for your work.

**Comment by sthibaul** [comment]
No news, thus closing
